src/twisted/test/test_process.py

```diff
@@ -2383,25 +2337,46 @@ class Win32UnicodeEnvironmentTests(unittest.TestCase):
     """
     Tests for Unicode environment on Windows
     """
-    goodKey = u'UNICODE'
-    goodValue = u'UNICODE'
+    goodKey = 'UNICODE'
+    goodKeyBytes = b'UNICODE'
+    goodValue = 'UNICODE'
+    goodValueBytes = b'UNICODE'
 
 
     def test_encodableUnicodeEnvironment(self):
         """
-        Test C{os.environ} (inherited by every subprocess on Windows) that
-        contains an ascii-encodable Unicode string. This is different from
-        passing Unicode environment explicitly to spawnProcess (which is not
-        supported on Python 2).
+        Test L{os.environ} (inherited by every subprocess on Windows) that
+        contains an ascii-encodable Unicode string.
         """
         os.environ[self.goodKey] = self.goodValue
         self.addCleanup(operator.delitem, os.environ, self.goodKey)
 
-        p = GetEnvironmentDictionary.run(reactor, [], properEnv)
-        def gotEnvironment(environ):
-            self.assertEqual(
-                environ[self.goodKey.encode('ascii')],
-                self.goodValue.encode('ascii'))
+        p = GetEnvironmentDictionary.run(reactor, [], os.environ)
+
+        def gotEnvironment(environb):
+            self.assertEqual(environb[self.goodKeyBytes], self.goodValueBytes)
+
+        return p.getResult().addCallback(gotEnvironment)
+
+
+    def test_UTF8StringInEnvironment(self):
+        """
+        L{os.environ} (inherited by every subprocess on Windows) can
+        contain a UTF-8 string.
```

Comment:
I am not sure I understand this docstring. ... will it handle Unicode key or value.

It looks like in this test, the environment contain Unicode (not UTF-8) and once  GetEnvironmentDictionary is called the value is automatically converted/encoded to UTF-8.

Does it mean that only the value can be Unicode?

I looks like both tests will have the same unicode, so my suggestion is to have a single test.

Reply:
This is documented in the Python 3 docs: https://docs.python.org/3/library/os.html#os.environ  `os.environ` contains `str` keys and values.  `str` is Unicode on Python 3.